File: Elixir/problem_6.ex
# Problem 6: Sum square difference
# https://projecteuler.net/problem=6

defmodule Problem6 do
  def sum_square_difference(n) do
    sum_of_squares = Enum.map(1..n, &(&1*&1)) |> Enum.sum
    round(:math.pow(Enum.sum(1..n),2) - sum_of_squares)
  end
end

# Result:
# > c "problem_6.ex"
# > Problem6.sum_square_difference(100)

File: R/problem_6.R
# Problem 6: Sum square difference
# https://projecteuler.net/problem=6

sum_square_difference <- function(n){
    sum_of_squares <- 0
    for(i in 1:n){
        sum_of_squares <- sum_of_squares + i^2
    }

    square_of_sum <- 0
    for(i in 1:n){
        square_of_sum <- square_of_sum + i
    }

    square_of_sum - sum_of_squares
}

# Result:
# > smallest_multiple(20)
# 232792560
